adapters/handlers/rest/configure_api.go

```diff
@@ -701,7 +702,12 @@ func configureAPI(api *operations.WeaviateAPI) http.Handler {
 	setupBackupHandlers(api, backupScheduler, appState.Metrics, appState.Logger)
 	setupNodesHandlers(api, appState.SchemaManager, appState.DB, appState)
 
-	grpcServer := createGrpcServer(appState)
+	var grpcInstrument []grpc.ServerOption
+	if appState.ServerConfig.Config.Monitoring.Enabled {
```

Comment:
hm, this optional observability looks weird to me and I see this is a common pattern across the repo.

If we really want to have an option to disable it, we could do that centrally by:
```go
prometheus.DefaultRegisterer = // set noop implementation
``` 

I will try to take a stab at this later

Reply:
The best option would be to avoid this global `prometheus.DefaultRegisterer` then no module init code would be required. Agree that this is not in the scope of this PR